crates/sui-bridge-indexer/src/main.rs

```diff
@@ -68,10 +64,21 @@ async fn main() -> Result<()> {
     );
     let indexer_meterics = BridgeIndexerMetrics::new(&registry);
 
-    // start indexing
-    start_processing_eth_events(&config).await?;
+    let eth_worker = EthBridgeWorker::new(
+        get_connection_pool(config.db_url.clone()),
+        indexer_meterics.clone(),
+        config,
+    );
+
+    // TODO: retry_with_max_elapsed_time
+
+    let unfinalized_handle = eth_worker.start_indexing_unfinalized_events();
+    let finalized_handle = eth_worker.start_indexing_finalized_events();
 
-    start_processing_sui_checkpoints(&config, indexer_meterics).await?;
+    let _ = tokio::try_join!(finalized_handle, unfinalized_handle);
```

Comment:
this won't work:
https://docs.rs/tokio/latest/tokio/macro.try_join.html
> Waits on multiple concurrent branches, returning when all branches complete with Ok(_) or on the first Err(_).

This means line 81 will never get a chance to run before the above two tasks finishes/errs


I think you want to move line 81 up and join all 3 tasks

Reply:
odd. so you're saying that the `start_processing_sui_checkpoints` function wont ever get called? It seems to be running in parallel to the eth_worker tasks....